packages/jsii-pacmak/lib/targets/version-utils.ts

```diff
@@ -55,10 +67,90 @@ export function toPythonVersionRange(semverRange: string): string {
     .join(', ');
 }
 
+/**
+ * Converts an original version number from the NPM convention to the target
+ * language's convention for expressing the same. For versions that do not
+ * include a prerelease identifier, this always returns the assembly version
+ * unmodified.
+ *
+ * @param assemblyVersion the assembly version being released
+ * @param target          the target language for which the verison is destined
+ *
+ * @returns the version that should be serialized
+ */
+export function toReleaseVersion(
+  assemblyVersion: string,
+  target: TargetName,
+): string {
+  const version = parse(assemblyVersion, { includePrerelease: true });
+  if (version == null) {
+    throw new Error(
+      `Unable to parse the provided assembly version: "${assemblyVersion}"`,
+    );
+  }
+  if (version.prerelease.length === 0) {
+    return assemblyVersion;
+  }
+  switch (target) {
+    case 'python':
+      // Python supports a limited set of identifiers... And we have a mapping table...
+      // https://packaging.python.org/guides/distributing-packages-using-setuptools/#pre-release-versioning
+      const [label, sequence, ...rest] = version.prerelease;
+      if (rest.length > 0) {
+        info(
+          `Unable to map prerelease identifier (in: ${assemblyVersion}) components to python: ${inspect(
+            version.prerelease,
+          )}`,
+        );
+        break;
+      }
+      if (!Number.isInteger(sequence)) {
+        info(
+          `Unable to map prerelease identifier (in: ${assemblyVersion}) to python, as sequence ${inspect(
+            sequence,
+          )} is not an integer`,
+        );
+        break;
+      }
+      switch (label) {
+        case 'dev':
+          return `${version.major}.${version.minor}.${version.patch}.dev${sequence}`;
+        case 'alpha':
+          return `${version.major}.${version.minor}.${version.patch}.a${sequence}`;
+        case 'beta':
+          return `${version.major}.${version.minor}.${version.patch}.b${sequence}`;
+        case 'rc':
+          return `${version.major}.${version.minor}.${version.patch}.rc${sequence}`;
```

Comment:
We should allow user overrides of the pre-release suffix, per language, in the `package.json`? 

Useful specifically for target package managers, like Python, that don't allow *any* pre-release identifier.
*use case:* customer releases pre-release version of their library in 1 language and now wants to expand to all, but has used a pre-release identifier that doesn't automatically map.

Reply:
Another reason this might be worth adding is that it's not clear what Python allows or disallows. [This spec](https://packaging.python.org/guides/distributing-packages-using-setuptools/#pre-release-versioning) states "may be published as" which is ambiguous.

And [PEP-440](https://www.python.org/dev/peps/pep-0440/#pre-release-spelling) states 'alpha' and 'beta' are allowed and, 'pre' represents release candidate. Is PEP-440 active?

Reply:
Might have found the first use case 😅

https://github.com/aws/constructs/blob/2a6bb57f02a22d844b71b97428f2545afb4f87ea/package.json#L126

Reply:
I would map `pre` to `dev` and the `constructs` problem would be gone?

I'm ambivalent about adding more knobs to override behaviors in the configuration files; because the onus is then on the user to know what is acceptable or not (so now, they have to go look at PEP-440, and get confused by how it seems to say something different from `pip`'s documentation....). I would rather add such a feature only once it has been demonstrated to be the only way to unblock a reasonable use-case.